src/jsn.erl

```diff
@@ -606,6 +656,65 @@ apply_selection(Selection, Element) ->
     erlang:error(badarg, [Selection, Element]).
 
 
+-spec from_proplist_impl(json_term(), format(), jsn_options()) -> json_term().
+%%------------------------------------------------------------------------------
+%% @private implement `from_proplist/2' with the format extracted
+%%------------------------------------------------------------------------------
+from_proplist_impl([{_,_}|_] = P, map, Opts) ->
+    lists:foldl(fun({Key, Value}, Acc) when ?IS_JSON_KEY(Key) ->
```

Comment:
@mrallen1 I was wondering if it would be more efficient, especially for larger proplists, to use

```erlang
M = maps:from_list(Plist),
maps:map(fun(Key, Value) when ?IS_JSON_KEY(Key) ->
             as_map(Value);
            (Key, Value) -> erlang:error(badarg, [Key, Value])
         end,
         M);
```

I haven't compared the performance of the 2 yet; just wondering if you've played with these two approaches before I bother with benchmarking.

Reply:
Sorry, I don't know. I think you're going to have to benchmark it to decide.

Reply:
Looks like the method in the comments works marginally better, and it appears (unsurprisingly) to be more pronounced with very long key-value pairs (especially over 32, I am guessing).

Since there's no real downside to doing it the second way (build map, then iterate over the map) for small objects, and a minor win for big ones, I will use that.